### PR TITLE
FINERACT-1241-elastic-web-hook

### PR DESCRIPTION
**Pull request by fynmanoj**
FINERACT-1241

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/commands/service/SynchronousCommandProcessingService.java [hunk 1]
```diff
@@ -37,6 +44,9 @@
 import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
 import org.apache.fineract.useradministration.domain.AppUser;
 import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/commands/service/SynchronousCommandProcessingService.java]: Joda Time support has just been removed.. use java.time.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/hooks/processor/ProcessorHelper.java [hunk 2]
```diff
@@ -121,4 +121,20 @@ public WebHookService createWebHookService(final String url) {
...
+        return new Callback() {
+
+            @Override
+            public void success(final Object o, final Response response) {
+                LOG.info("URL : {} \tStatus : {}", url, response.getStatus());
+            }
+
+            @Override
+            public void failure(final RetrofitError retrofitError) {
+                LOG.info("{}", retrofitError.getMessage());
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/hooks/processor/ProcessorHelper.java]: LOG.error() is probably better here?

**Review by vorburger — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/commands/service/SynchronousCommandProcessingService.java [hunk 3]
```diff
@@ -205,18 +225,80 @@ public boolean validateCommand(final CommandWrapper commandWrapper, final AppUse
...
+                reqmap.put("request", myMap);
+                if (result instanceof CommandProcessingResult) {
+                    CommandProcessingResult resultCopy = CommandProcessingResult
+                            .fromCommandProcessingResult((CommandProcessingResult) result);
+                    if (resultCopy.hasChanges() && resultCopy.getChanges().containsKey("receipt")
+                            && resultCopy.getChanges().get("receipt") instanceof String) {
+                        String receipt = (String) resultCopy.getChanges().get("receipt");
+                        if (receipt.startsWith("{")) {
+                            try {
+                                resultCopy.getChanges().put("receiptObj", gson.fromJson(receipt, JsonObject.class));
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/commands/service/SynchronousCommandProcessingService.java]: Don't think this receiptObj exists in any APIs.

**Review by avikganguly01 — APPROVED** [review]
Requested changes addressed.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/hooks/processor/ElasticSearchHookProcessor.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import org.springframework.stereotype.Service;
+import retrofit.Callback;
+
+@Service
+public class ElasticSearchHookProcessor implements HookProcessor {
+
+    @Autowired
+    private ProcessorHelper processorHelper;
+
+    private PrintWriter out = null;
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/hooks/processor/ElasticSearchHookProcessor.java:43]: This looks like it's not multi thread safe? Or am I missing anything?

**Comment by vorburger** [comment]
@fynmanoj @avikganguly01 I had to revert this PR (in #1473), further details in FINERACT-1241. Will you re-raise it?

**Pull request merged**